src/servers/grpc_server.cc

```diff
@@ -2305,13 +2341,26 @@ InferResponseAlloc(
   // will be creating and using just one response object.
   inference::ModelInferResponse* response =
       payload->response_queue_->GetNonDecoupledResponse();
-  return ResponseAllocatorHelper<
-      AllocPayload<inference::ModelInferResponse>::TensorShmMap>(
+  return ResponseAllocatorHelper(
       allocator, tensor_name, byte_size, preferred_memory_type,
       preferred_memory_type_id, response, payload->shm_map_, buffer,
       buffer_userp, actual_memory_type, actual_memory_type_id);
 }
 
+TRITONSERVER_Error*
+OutputBufferQuery(
```

Comment:
Add comment here and before InferResponseAlloc above.
"Make sure to keep InferResponseAlloc and OutputBufferQuery logic in sync"